src/background/actions/assessment-action-creator.ts

```diff
@@ -152,137 +154,159 @@ export class AssessmentActionCreator {
+        await this.assessmentActions.updateTargetTabId.invoke(tabId);
+        await this.assessmentActions.changeRequirementStatus.invoke(payload);
     };
 
-    private onUndoChangeManualRequirementStatus = (
+    private onUndoChangeManualRequirementStatus = async (
         payload: ChangeRequirementStatusPayload,
-    ): void => {
+    ): Promise<void> => {
         const eventName = TelemetryEvents.UNDO_REQUIREMENT_STATUS_CHANGE;
         this.telemetryEventHandler.publishTelemetry(eventName, payload);
-        this.assessmentActions.undoRequirementStatusChange.invoke(payload);
+        await this.assessmentActions.undoRequirementStatusChange.invoke(payload);
     };
 
-    private onUndoAssessmentInstanceStatusChange = (
+    private onUndoAssessmentInstanceStatusChange = async (
         payload: AssessmentActionInstancePayload,
-    ): void => {
+    ): Promise<void> => {
         const eventName = TelemetryEvents.UNDO_TEST_STATUS_CHANGE;
         this.telemetryEventHandler.publishTelemetry(eventName, payload);
-        this.assessmentActions.undoInstanceStatusChange.invoke(payload);
+        await this.assessmentActions.undoInstanceStatusChange.invoke(payload);
     };
 
-    private onChangeAssessmentInstanceStatus = (
+    private onChangeAssessmentInstanceStatus = async (
         payload: ChangeInstanceStatusPayload,
         tabId: number,
-    ): void => {
+    ): Promise<void> => {
         const eventName = TelemetryEvents.CHANGE_INSTANCE_STATUS;
         this.telemetryEventHandler.publishTelemetry(eventName, payload);
-        this.assessmentActions.updateTargetTabId.invoke(tabId);
-        this.assessmentActions.changeInstanceStatus.invoke(payload);
+        await this.assessmentActions.updateTargetTabId.invoke(tabId);
+        await this.assessmentActions.changeInstanceStatus.invoke(payload);
     };
 
-    private onChangeAssessmentVisualizationState = (
+    private onChangeAssessmentVisualizationState = async (
         payload: ChangeInstanceSelectionPayload,
-    ): void => {
+    ): Promise<void> => {
         const eventName = TelemetryEvents.CHANGE_ASSESSMENT_VISUALIZATION_STATUS;
         this.telemetryEventHandler.publishTelemetry(eventName, payload);
-        this.assessmentActions.changeAssessmentVisualizationState.invoke(payload);
+        await this.assessmentActions.changeAssessmentVisualizationState.invoke(payload);
     };
 
-    private onChangeVisualizationStateForAll = (payload: ChangeInstanceSelectionPayload): void => {
+    private onChangeVisualizationStateForAll = async (
+        payload: ChangeInstanceSelectionPayload,
+    ): Promise<void> => {
         const eventName = TelemetryEvents.CHANGE_ASSESSMENT_VISUALIZATION_STATUS_FOR_ALL;
         this.telemetryEventHandler.publishTelemetry(eventName, payload);
-        this.assessmentActions.changeAssessmentVisualizationStateForAll.invoke(payload);
+        await this.assessmentActions.changeAssessmentVisualizationStateForAll.invoke(payload);
```

Comment:
Why are some of the invoke calls using the executing scope and some aren't? Can we either standardize across the file or comment what the difference is?